catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -119,26 +125,35 @@ public void initialize(
       Map<String, String> config, CatalogInfo info, HasPropertyMetadata propertiesMetadata)
       throws RuntimeException {
     this.propertiesMetadata = propertiesMetadata;
-    // Initialize Hadoop Configuration.
-    this.conf = config;
-    this.hadoopConf = new Configuration();
     this.catalogInfo = info;
-    Map<String, String> bypassConfigs =
-        config.entrySet().stream()
-            .filter(e -> e.getKey().startsWith(CATALOG_BYPASS_PREFIX))
-            .collect(
-                Collectors.toMap(
-                    e -> e.getKey().substring(CATALOG_BYPASS_PREFIX.length()),
-                    Map.Entry::getValue));
-    bypassConfigs.forEach(hadoopConf::set);
+
+    this.conf = config;
+
+    String fileSystemProviders =
+        (String)
+            propertiesMetadata
+                .catalogPropertiesMetadata()
+                .getOrDefault(
+                    config, HadoopCatalogPropertiesMetadata.FILESYSTEM_PROVIDERS_CLASSNAMES);
+    FileSystemUtils.initFileSystemProviders(fileSystemProviders, fileSystemProvidersMap);
+
+    String defaultFileSystemProviderClassName =
+        (String)
+            propertiesMetadata
+                .catalogPropertiesMetadata()
+                .getOrDefault(
+                    config, HadoopCatalogPropertiesMetadata.DEFAULT_FS_PROVIDER_CLASSNAME);
+    this.defaultFileSystemProviderScheme =
+        StringUtils.isNotBlank(defaultFileSystemProviderClassName)
+            ? FileSystemUtils.getSchemeByFileSystemProvider(
+                defaultFileSystemProviderClassName, fileSystemProvidersMap)
+            : LOCAL_FILE_SCHEME;
```

Comment:
You can create the default fs provider and get scheme from default fs provider, no need to use this logic.

Reply:
If we use the service load mechanism to load file system providers, I can't get why we need to provide the configuration `filesystem-providers`, please see below

```java

// fileSystemProviders is useless if we use ServiceLoader 
  public static Map<String, FileSystemProvider> getFileSystemProviders(String fileSystemProviders) {
    Map<String, FileSystemProvider> resultMap = Maps.newHashMap();
    ServiceLoader<FileSystemProvider> fileSystemProvidersLoader =
        ServiceLoader.load(FileSystemProvider.class);
    fileSystemProvidersLoader.forEach(
        fileSystemProvider -> resultMap.put(fileSystemProvider.name(), fileSystemProvider));
    return resultMap;
  }
```

then the only configuration we need to provide is `default-filesystem-provider`,  @jerryshao what do you think about it, is it okay for you?